src/pymortests/algorithms/pod.py

```diff
@@ -4,24 +4,45 @@
 
 import numpy as np
 import pytest
+from hypothesis import given, assume, reproduce_failure, settings
+from hypothesis.strategies import sampled_from
 
 from pymor.algorithms.basic import almost_equal
+from pymor.algorithms.gram_schmidt import gram_schmidt
 from pymor.algorithms.pod import pod
+from pymor.algorithms.basic import contains_zero_vector
 from pymortests.fixtures.operator import operator_with_arrays_and_products
-from pymortests.fixtures.vectorarray import vector_array, vector_array_without_reserve
+from pymortests.strategies import vector_arrays, implementations
 
 methods = ['method_of_snapshots', 'qr_svd']
 
 
-@pytest.mark.parametrize('method', methods)
+@settings(deadline=None)
+@implementations(method=sampled_from(methods))
 def test_pod(vector_array, method):
     A = vector_array
+    product = None
     print(type(A))
     print(A.dim, len(A))
+    # TODO assumption here masks a potential issue with the algorithm
+    #      where it fails in internal lapack instead of a proper error
+    assume(len(A) > 1 or A.dim > 1)
+    assume(not contains_zero_vector(A, rtol=1e-13, atol=1e-13))
 
     B = A.copy()
-    U, s = pod(A, method=method)
+    # we run the pod with infinite tol so we can manually check gram_schidt and adjust assertions
```

Comment:
I still don't understand what you are trying to do here ..

Reply:
If the pod runs with a finite (default) tolerance we might get a shorter `U` than expected. So I need to manually run `gram_schmidt` afterwards to be able to adjust for that.
Before hypothesis the input data was so that the orthogonalization in pod never truncated the vector_array, so the check for equal length always succeeded.

Reply:
Oh, so if `U` and `s` can actually be of a different length, I would consider this as a bug. They are supposed to be of equal lengths.

Reply:
Huh, I hadn't actually considered that tbh. I only considered that we simply hadn't input "bad enough" data yet.

Reply:
Made a note: https://github.com/pymor/pymor/issues/961